bin/ROC_Efficiency.cc

```diff
@@ -0,0 +1,235 @@
+{
+  public:
+    HitCounter () {
+      NFiducial[0] = 0;
+      NFiducial[1] = 0;
+      NFiducial[2] = 0;
+      NFiducialAndHit[0] = 0;
+      NFiducialAndHit[1] = 0;
+      NFiducialAndHit[2] = 0;
+    }
+    ~HitCounter () {}
+
+    int NFiducial[3];
+    int NFiducialAndHit[3];
+};
+
+
+
+// FUNCTION DEFINITIONS HERE
+int TrackingEfficiency (std::string const, std::string const, std::string const);
+
+
+
+
+
+
+
+// CODE BELOW
+
+int TrackingEfficiency (std::string const DataFileName, std::string const GainCalFileName, std::string const AlignmentFileName)
+{
+  std::cout << "DataFileName:      " << DataFileName << std::endl;
+  std::cout << "AlignmentFileName: " << AlignmentFileName << std::endl;
+
+  // Set some basic style for plots
+  PLTU::SetStyle();
+
+  // Grab the plt event reader
+  PLTEvent Event(DataFileName, GainCalFileName, AlignmentFileName);
+
+  PLTPlane::FiducialRegion FidRegionHits  = PLTPlane::kFiducialRegion_Diamond;
+  PLTPlane::FiducialRegion FidRegionTrack = PLTPlane::kFiducialRegion_m5_m5;//<-Original
+  Event.SetPlaneFiducialRegion(FidRegionHits);
+  Event.SetPlaneClustering(PLTPlane::kClustering_AllTouching,PLTPlane::kFiducialRegion_All);
+
+  PLTAlignment Alignment;
+  Alignment.ReadAlignmentFile(AlignmentFileName);
+
+  // Map for tracking efficiency
+  std::map<uint32_t, std::map<int, HitCounter> > HC;
+  
+  float const PixelDist = 5;//<-Original
+  float slope_x_low = 0.0 - 0.01;
+  float slope_y_low = 0.027 - 0.01;
+  float slope_x_high = 0.0 + 0.01;
+  float slope_y_high = 0.027 + 0.01;
+
+  uint32_t date = 0;
+  uint32_t five_min = 5 * 60 * 1000; 
+
+  // Loop over all events in file
+  for (int ientry = 0; Event.GetNextEvent() >= 0; ++ientry) {
+//    if (ientry > 50000) break;
+    if (ientry == 0){
+      date = Event.Time();
+    }
+
+    if (ientry % 10000 == 0) {
+      std::cout << "Processing entry: " << ientry << std::endl;
+    }
+
+    // Loop over all planes with hits in event
+    for (size_t it = 0; it != Event.NTelescopes(); ++it) {
+      PLTTelescope* Telescope = Event.Telescope(it);
+
+      int    const Channel = Telescope->Channel();
+      size_t const NPlanes = Telescope->NPlanes();
+
+      // make them clean events
+      if (Telescope->NHitPlanes() < 2 || Telescope->NHitPlanes() != Telescope->NClusters()) {
+        continue;
+      }
+
+
+
+      PLTPlane* Plane[3] = {0x0, 0x0, 0x0};
+      for (size_t ip = 0; ip != NPlanes; ++ip) {
+        Plane[ Telescope->Plane(ip)->ROC() ] = Telescope->Plane(ip);
+      }
+
+      // To construct 4 tracks.. one testing each plane.. and one using all planes if it be.
+      PLTTrack Tracks[4];
+
+      // If it has all 3 planes that'll be number 0
+      if (Plane[0]->NClusters() && Plane[1]->NClusters() && Plane[2]->NClusters()) {
+        Tracks[0].AddCluster(Plane[0]->Cluster(0));
+        Tracks[0].AddCluster(Plane[1]->Cluster(0));
+        Tracks[0].AddCluster(Plane[2]->Cluster(0));
+        Tracks[0].MakeTrack(Alignment);
+      }
```

Comment:
since you're no longer making the plots for the track on all three planes, this one doesn't actually get used ... you could just get rid of it.